armi/interfaces.py

```diff
@@ -77,6 +81,137 @@ class STACK_ORDER:  # pylint: disable=invalid-name, too-few-public-methods
     POSTPROCESSING = BOOKKEEPING + 1
 
 
+class TightCoupler:
+    """
+    Data structure that defines tight coupling attributes that are implemented
+    within an Interface and called upon when ``interactCoupled`` is called.
+
+    Parameters
+    ----------
+    param : str
+        The name of a parameter defined in the ARMI Reactor model.
+
+    tolerance : float
+        Defines the allowable error, epsilon, between the current previous
+        parameter value(s) to determine if the selected coupling parameter has
+        been converged.
+
+    maxIters : int
+        Maximum number of tight coupling iterations allowed
+    """
+
+    _SUPPORTED_TYPES = [float, int, list, numpy.ndarray]
+
+    def __init__(self, param, tolerance, maxIters):
```

Comment:
One thing that I've noticed is that every time a `TightCoupler` class is instantiated, it is necessary to pass in `maxIters`. However, this value should be the same for all instances of the `TightCoupler`, because it comes from the case settings. 

This is just a little unideal, because it somehow could trigger the coupling iterations to stop early if one of the interfaces is supplied with a `maxIters` smaller than the others. 

This could be resolved if the tight coupling mechanics were pulled off of the interfaces into somewhere higher, perhaps the operator or into its own interface. I put a comment such as this on the PR page, but I'm gonna put another one again with some more thoughts.

Reply:
Passing in an integer seems like the lightest-possible-weight solution to me.

I like that not ALL of our code has to directly call the CaseSettings object.  It makes the code more reusable, and flexible. And helps us toward our [Roadmap](https://github.com/terrapower/armi/wiki/Roadmap) of not having the "Operator is King" mentality.